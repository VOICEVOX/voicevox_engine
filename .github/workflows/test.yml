name: test

on:
  push:
  pull_request:
    branches:
      - '**'

jobs:
  build:
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [ubuntu-latest, windows-latest]  # [ubuntu-latest, macos-latest, windows-latest]
        python: [3.7]
        include:
        - os: ubuntu-latest
          path: ~/.cache/pip
        # - os: macos-latest
        #   path: ~/Library/Caches/pip
        - os: windows-latest
          path: ~\AppData\Local\pip\Cache

    steps:
      - uses: actions/checkout@v2

      - name: Set up Python ${{ matrix.python }}
        uses: actions/setup-python@v2
        with:
          python-version: ${{ matrix.python }}

      - uses: actions/cache@v2
        id: pip-cache
        with:
          path: ${{ matrix.path }}
          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt', '**/requirements-test.txt', '**/requirements-dev.txt') }}
          restore-keys: |
            ${{ runner.os }}-pip-

      - name: Install libraries for ubuntu
        if: matrix.os == 'ubuntu-latest'
        run: sudo apt-get install libsndfile1-dev

      - name: Install dependencies
        run: |
          python -m pip install --upgrade pip setuptools wheel
          pip install -r requirements-test.txt

      - run: pysen run lint

      - name: Run pytest and get coverage
        run: |
          coverage run --omit=test/* -m pytest

      - name: Submit coverage to Coveralls
        if: matrix.os == 'ubuntu-latest'
        run: coveralls --service=github
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

      - name: Create coverage result
        if: github.event_name == 'pull_request' && matrix.os == 'ubuntu-latest'
        run: |
          coverage report > report.txt

      - name: Comment coverage result to Pull Requests
        if: github.event_name == 'pull_request' && matrix.os == 'ubuntu-latest'
        uses: actions/github-script@v5.0.0
        with:
          github-token: ${{ secrets.TOKEN_THAT_CAN_GITHUB_REPO_ACCESS }}
          script: |
            const fs = require('fs')
            const baseReport = fs.readFileSync('report.txt', 'utf8').toString().split('\n')
            let report = ''
            for (let i = 0; i < baseReport.length; i++) {
              const line = baseReport[i].split(' ').filter(v => v)
              if (i === 1 && line.length === 1) {
                report += "|:---|---:|---:|---:|\n"
              } else if (line.length === 1) {
                continue
              } else {
                if (i !== 0 && line.length === 4) {
                  const parcent = Number(line[3].replace("%", ""))
                  let color = 'green'
                  if (parcent < 50) {
                    color = 'red'
                  } else if (parcent < 90) {
                    color = 'orange'
                  }
                  line[3] = `![coverage-${parcent}%](https://img.shields.io/badge/coverage-${parcent}%25-${color}.svg)`
                }
                report += "|" + line.join("|") + "|\n"
              }
              if (line[0] === 'TOTAL') break
            }

            github.rest.issues.createComment({
              issue_number: context.issue.number,
              owner: context.repo.owner,
              repo: context.repo.repo,
              body: `## Coverage Result\n\n${report}`
            })

      - name: Check licenses
        shell: bash
        run: |
          python -m venv venv

          if [ -d "venv/Scripts" ]; then
            source venv/Scripts/activate
          else
            source venv/bin/activate
          fi

          pip install -r requirements-dev.txt
          python generate_licenses.py > /dev/null
