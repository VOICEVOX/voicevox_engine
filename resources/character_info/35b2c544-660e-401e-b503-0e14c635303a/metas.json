{
  "supported_features": { "permitted_synthesis_morphing": "NOTHING" }
}
