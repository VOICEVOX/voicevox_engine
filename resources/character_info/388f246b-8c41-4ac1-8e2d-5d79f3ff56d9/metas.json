{
  "supported_features": { "permitted_synthesis_morphing": "SELF_ONLY" }
}
