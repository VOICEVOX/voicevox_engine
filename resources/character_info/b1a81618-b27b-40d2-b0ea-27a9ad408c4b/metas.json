{
  "supported_features": { "permitted_synthesis_morphing": "ALL" }
}
