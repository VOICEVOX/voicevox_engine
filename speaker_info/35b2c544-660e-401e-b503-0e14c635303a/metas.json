{
    "speakerName": "dummy3",
    "speakerUuid": "dummy3uuid",
    "supportedFeatures": {"permitedSynthesisMorphing": "NOTHING"},
    "styles": [
        {
            "styleId": 8
        }
    ]
}