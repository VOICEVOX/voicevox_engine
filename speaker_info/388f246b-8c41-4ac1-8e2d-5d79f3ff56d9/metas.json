{
    "speakerName": "dummy2",
    "speakerUuid": "dummy2uuid",
    "supportedFeatures": {"permitedSynthesisMorphing": "SELF_ONLY"},
    "styles": [
        {
            "styleId": 3
        },
        {
            "styleName": "style2",
            "styleId": 1
        },
        {
            "styleName": "style3",
            "styleId": 7
        },
        {
            "styleName": "style4",
            "styleId": 5
        }
    ]
}