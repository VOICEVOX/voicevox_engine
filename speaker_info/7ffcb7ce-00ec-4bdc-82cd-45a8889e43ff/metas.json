{
    "supported_features": {
        "permitted_synthesis_morphing": "SELF_ONLY"
    },
    "range": [
        {
            "style_id": 2,
            "low": 5.16,
            "high": 6.2
        },
        {
            "style_id": 0,
            "low": 5.21,
            "high": 6.13
        }
    ]
}