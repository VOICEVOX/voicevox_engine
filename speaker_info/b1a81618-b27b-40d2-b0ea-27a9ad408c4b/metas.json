{
    "speakerName": "dummy4",
    "speakerUuid": "dummy4uuid",
    "supportedFeatures": {"permitedSynthesisMorphing": "ALL"},
    "styles": [
        {
            "styleId": 9
        }
    ]
}