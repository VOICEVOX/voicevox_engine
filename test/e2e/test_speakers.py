"""
話者・歌手のテスト。
TODO: 話者と歌手の両ドメイン共通のドメイン用語を定め、このテストファイル名を変更する。
"""

from test.utility import hash_long_string

from fastapi.testclient import TestClient
from pydantic import parse_obj_as
from syrupy.assertion import SnapshotAssertion

from voicevox_engine.metas.Metas import Speaker


def test_話者一覧が取得できる(
    client: TestClient, snapshot_json: SnapshotAssertion
) -> None:
    response = client.get("/speakers")
    assert response.status_code == 200
    assert snapshot_json == response.json()


def test_話者の情報を取得できる(
    client: TestClient, snapshot_json: SnapshotAssertion
) -> None:
    speakers = parse_obj_as(list[Speaker], client.get("/speakers").json())
    for speaker in speakers:
        response = client.get(
            "/speaker_info", params={"speaker_uuid": speaker.speaker_uuid}
        )
        assert snapshot_json(
            name=speaker.speaker_uuid,
        ) == hash_long_string(response.json())


def test_歌手一覧が取得できる(
    client: TestClient, snapshot_json: SnapshotAssertion
) -> None:
    response = client.get("/singers")
    assert response.status_code == 200
    assert snapshot_json == response.json()


def test_歌手の情報を取得できる(
    client: TestClient, snapshot_json: SnapshotAssertion
) -> None:
    singers = parse_obj_as(list[Speaker], client.get("/singers").json())
    for singer in singers:
        response = client.get(
            "/singer_info", params={"speaker_uuid": singer.speaker_uuid}
        )
        assert snapshot_json(
            name=singer.speaker_uuid,
        ) == hash_long_string(response.json())
