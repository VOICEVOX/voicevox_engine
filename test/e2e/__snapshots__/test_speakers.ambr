# serializer version: 1
# name: test_歌手の情報をURLで取得できる[388f246b-8c41-4ac1-8e2d-5d79f3ff56d9_5_icon]
  'MD5:517ba089e0b03f8868af2ce956f7699d'
# ---
# name: test_歌手の情報をURLで取得できる[388f246b-8c41-4ac1-8e2d-5d79f3ff56d9_5_voice_sample_0]
  'MD5:517ba089e0b03f8868af2ce956f7699d'
# ---
# name: test_歌手の情報をURLで取得できる[388f246b-8c41-4ac1-8e2d-5d79f3ff56d9_5_voice_sample_1]
  'MD5:517ba089e0b03f8868af2ce956f7699d'
# ---
# name: test_歌手の情報をURLで取得できる[388f246b-8c41-4ac1-8e2d-5d79f3ff56d9_5_voice_sample_2]
  'MD5:517ba089e0b03f8868af2ce956f7699d'
# ---
# name: test_歌手の情報をURLで取得できる[388f246b-8c41-4ac1-8e2d-5d79f3ff56d9_7_icon]
  'MD5:562ad0f61ca6dd81e89a4479f97dcd9f'
# ---
# name: test_歌手の情報をURLで取得できる[388f246b-8c41-4ac1-8e2d-5d79f3ff56d9_7_voice_sample_0]
  'MD5:562ad0f61ca6dd81e89a4479f97dcd9f'
# ---
# name: test_歌手の情報をURLで取得できる[388f246b-8c41-4ac1-8e2d-5d79f3ff56d9_7_voice_sample_1]
  'MD5:562ad0f61ca6dd81e89a4479f97dcd9f'
# ---
# name: test_歌手の情報をURLで取得できる[388f246b-8c41-4ac1-8e2d-5d79f3ff56d9_7_voice_sample_2]
  'MD5:562ad0f61ca6dd81e89a4479f97dcd9f'
# ---
# name: test_歌手の情報をURLで取得できる[388f246b-8c41-4ac1-8e2d-5d79f3ff56d9_portrait]
  'MD5:27777cb0883c98cd9870707005bf1faf'
# ---
# name: test_歌手の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_4_icon]
  'MD5:c6a8ddea789d8115372db31b4a76d2aa'
# ---
# name: test_歌手の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_4_portrait]
  'MD5:c6a8ddea789d8115372db31b4a76d2aa'
# ---
# name: test_歌手の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_4_voice_sample_0]
  'MD5:c6a8ddea789d8115372db31b4a76d2aa'
# ---
# name: test_歌手の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_4_voice_sample_1]
  'MD5:c6a8ddea789d8115372db31b4a76d2aa'
# ---
# name: test_歌手の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_4_voice_sample_2]
  'MD5:c6a8ddea789d8115372db31b4a76d2aa'
# ---
# name: test_歌手の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_6_icon]
  'MD5:d2c7c85a9919372ef62ad49786bd7fc4'
# ---
# name: test_歌手の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_6_portrait]
  'MD5:d2c7c85a9919372ef62ad49786bd7fc4'
# ---
# name: test_歌手の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_6_voice_sample_0]
  'MD5:d2c7c85a9919372ef62ad49786bd7fc4'
# ---
# name: test_歌手の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_6_voice_sample_1]
  'MD5:d2c7c85a9919372ef62ad49786bd7fc4'
# ---
# name: test_歌手の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_6_voice_sample_2]
  'MD5:d2c7c85a9919372ef62ad49786bd7fc4'
# ---
# name: test_歌手の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_portrait]
  'MD5:78228648fe527ee23597b66db59a4f94'
# ---
# name: test_歌手の情報をURLで取得できる[b1a81618-b27b-40d2-b0ea-27a9ad408c4b_9_icon]
  'MD5:1e3d0a6e88264b1b99839aeb7c7cbe1f'
# ---
# name: test_歌手の情報をURLで取得できる[b1a81618-b27b-40d2-b0ea-27a9ad408c4b_9_voice_sample_0]
  'MD5:1e3d0a6e88264b1b99839aeb7c7cbe1f'
# ---
# name: test_歌手の情報をURLで取得できる[b1a81618-b27b-40d2-b0ea-27a9ad408c4b_9_voice_sample_1]
  'MD5:1e3d0a6e88264b1b99839aeb7c7cbe1f'
# ---
# name: test_歌手の情報をURLで取得できる[b1a81618-b27b-40d2-b0ea-27a9ad408c4b_9_voice_sample_2]
  'MD5:1e3d0a6e88264b1b99839aeb7c7cbe1f'
# ---
# name: test_歌手の情報をURLで取得できる[b1a81618-b27b-40d2-b0ea-27a9ad408c4b_portrait]
  'MD5:381ba07cbcad95f1c99109f5f6d096a8'
# ---
# name: test_話者の情報をURLで取得できる[35b2c544-660e-401e-b503-0e14c635303a_8_icon]
  'MD5:1fe576e75458c752cfeecc1e93a29886'
# ---
# name: test_話者の情報をURLで取得できる[35b2c544-660e-401e-b503-0e14c635303a_8_portrait]
  'MD5:1fe576e75458c752cfeecc1e93a29886'
# ---
# name: test_話者の情報をURLで取得できる[35b2c544-660e-401e-b503-0e14c635303a_8_voice_sample_0]
  'MD5:1fe576e75458c752cfeecc1e93a29886'
# ---
# name: test_話者の情報をURLで取得できる[35b2c544-660e-401e-b503-0e14c635303a_8_voice_sample_1]
  'MD5:1fe576e75458c752cfeecc1e93a29886'
# ---
# name: test_話者の情報をURLで取得できる[35b2c544-660e-401e-b503-0e14c635303a_8_voice_sample_2]
  'MD5:1fe576e75458c752cfeecc1e93a29886'
# ---
# name: test_話者の情報をURLで取得できる[35b2c544-660e-401e-b503-0e14c635303a_portrait]
  'MD5:bd6cf66dcc652f56892b14b423f6f37c'
# ---
# name: test_話者の情報をURLで取得できる[388f246b-8c41-4ac1-8e2d-5d79f3ff56d9_1_icon]
  'MD5:becb1cc2aaf82623a13a1250a39d7393'
# ---
# name: test_話者の情報をURLで取得できる[388f246b-8c41-4ac1-8e2d-5d79f3ff56d9_1_voice_sample_0]
  'MD5:becb1cc2aaf82623a13a1250a39d7393'
# ---
# name: test_話者の情報をURLで取得できる[388f246b-8c41-4ac1-8e2d-5d79f3ff56d9_1_voice_sample_1]
  'MD5:becb1cc2aaf82623a13a1250a39d7393'
# ---
# name: test_話者の情報をURLで取得できる[388f246b-8c41-4ac1-8e2d-5d79f3ff56d9_1_voice_sample_2]
  'MD5:becb1cc2aaf82623a13a1250a39d7393'
# ---
# name: test_話者の情報をURLで取得できる[388f246b-8c41-4ac1-8e2d-5d79f3ff56d9_3_icon]
  'MD5:9a3690c368cd9a4ecb1940ff9eb2c955'
# ---
# name: test_話者の情報をURLで取得できる[388f246b-8c41-4ac1-8e2d-5d79f3ff56d9_3_portrait]
  'MD5:9a3690c368cd9a4ecb1940ff9eb2c955'
# ---
# name: test_話者の情報をURLで取得できる[388f246b-8c41-4ac1-8e2d-5d79f3ff56d9_3_voice_sample_0]
  'MD5:9a3690c368cd9a4ecb1940ff9eb2c955'
# ---
# name: test_話者の情報をURLで取得できる[388f246b-8c41-4ac1-8e2d-5d79f3ff56d9_3_voice_sample_1]
  'MD5:9a3690c368cd9a4ecb1940ff9eb2c955'
# ---
# name: test_話者の情報をURLで取得できる[388f246b-8c41-4ac1-8e2d-5d79f3ff56d9_3_voice_sample_2]
  'MD5:9a3690c368cd9a4ecb1940ff9eb2c955'
# ---
# name: test_話者の情報をURLで取得できる[388f246b-8c41-4ac1-8e2d-5d79f3ff56d9_portrait]
  'MD5:27777cb0883c98cd9870707005bf1faf'
# ---
# name: test_話者の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_0_icon]
  'MD5:1f1da5f25968c638a783bf6ba9df9420'
# ---
# name: test_話者の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_0_portrait]
  'MD5:1f1da5f25968c638a783bf6ba9df9420'
# ---
# name: test_話者の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_0_voice_sample_0]
  'MD5:1f1da5f25968c638a783bf6ba9df9420'
# ---
# name: test_話者の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_0_voice_sample_1]
  'MD5:1f1da5f25968c638a783bf6ba9df9420'
# ---
# name: test_話者の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_0_voice_sample_2]
  'MD5:1f1da5f25968c638a783bf6ba9df9420'
# ---
# name: test_話者の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_2_icon]
  'MD5:0b158046338f60a53e9afdb7797c5864'
# ---
# name: test_話者の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_2_portrait]
  'MD5:0b158046338f60a53e9afdb7797c5864'
# ---
# name: test_話者の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_2_voice_sample_0]
  'MD5:0b158046338f60a53e9afdb7797c5864'
# ---
# name: test_話者の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_2_voice_sample_1]
  'MD5:0b158046338f60a53e9afdb7797c5864'
# ---
# name: test_話者の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_2_voice_sample_2]
  'MD5:0b158046338f60a53e9afdb7797c5864'
# ---
# name: test_話者の情報をURLで取得できる[7ffcb7ce-00ec-4bdc-82cd-45a8889e43ff_portrait]
  'MD5:78228648fe527ee23597b66db59a4f94'
# ---
